controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java

```diff
@@ -772,6 +773,10 @@ protected void executeSingleTargetChild(PathAddress base, PathElement currentEle
             if (resource.hasChild(currentElement) || (nr != null && nr.isRemote())) {
                 safeExecute(next, newRemaining, context, nr, ignoreMissing);
             }
+            //if we are on the wrong host no need to do anything
+            if(!resource.hasChild(currentElement) && !HOST.equals(currentElement.getKey()) && (newRemaining != null && newRemaining.size() > 0)) {
+               throw new Resource.NoSuchResourceException(currentElement);
+            }
```

Comment:
1) Shouldn't this be an 'else if'?

2) I don't get the '&& !HOST.equals(currentElement.getKey())' bit. Why not throw an exception? Is this about someone connecting the a slave HC and doing /host=master/...:read-resource? So asking about a real host but one that the slave can't know about? If so I think the exception is fine.

3) Re: '&& (newRemaining != null && newRemaining.size() > 0)' -- looking at the only caller of this method, neither of those conditions will be true, so let's just javadoc that fact in the executeSingleTargetChild javadoc and drop this check.